### PR TITLE
Add GasLeft syscall

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/1508

### CODE REVIEW

**Comment by erikzhang** [comment]
In test mode, you don't have GAS.

**Comment by shargon** [comment]
Maybe we can `return false` if it's test mode

**Comment by erikzhang** [comment]
Maybe just push `-1`, and return `true`.

File: src/neo/SmartContract/InteropService.Runtime.cs [hunk 1]
```diff
@@ -29,6 +29,7 @@ public static class Runtime
             public static readonly InteropDescriptor Log = Register("System.Runtime.Log", Runtime_Log, 0_01000000, TriggerType.All, CallFlags.AllowNotify);
             public static readonly InteropDescriptor Notify = Register("System.Runtime.Notify", Runtime_Notify, 0_01000000, TriggerType.All, CallFlags.AllowNotify);
             public static readonly InteropDescriptor GetNotifications = Register("System.Runtime.GetNotifications", Runtime_GetNotifications, 0_00010000, TriggerType.All, CallFlags.None);
+            public static readonly InteropDescriptor GasLeft = Register("System.Runtime.GasLeft", Runtime_GasLeft, 0_00000400, TriggerType.All, CallFlags.None);
```
Comment (shargon) [inline src/neo/SmartContract/InteropService.Runtime.cs]: @erikzhang Did you checked the cost? It's the same of `GetInvocationCounter` I don't know if it's the correct

Comment (erikzhang) [reply src/neo/SmartContract/InteropService.Runtime.cs]: I think it's fine.

**Comment by ProDog** [comment]
I added this method to the Neo.SmartContract.Framework and tested,it works.
in the Neo.SmartContract.Framework.Services.Neo.Runtime:

image

in the contract:

image


invoke result:
test mode:

image

true gas left:

image

**Pull request merged**